### DISCUSSION TITLE
📚 Add hook to "destroy" or "unload" the Loop Carousel instance

### DISCUSSION THREAD

**Discussion by rmorse** [elementor/elementor#22894] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

This is borderline bug / feature request.

I'm trying to integrate our plugin with the Loop Carousel (we've already successfully integrated with the [Loop Grid](https://searchandfilter.com/documentation/3rd-party/elementor/))

However I'm having issue with the Loop Carousel as its throwing an error whenever we update the HTML.

I think we need a way to "destroy" or "unload" the Loop Carousel JS.

Our integration with the Loop Grid works something like this:

1. A user searches with our plugin
2. We load the results page via JS request
3. Extract the results HTML (loop grid) and replace it on the current page
4. Then we run `window.elementorFrontend.elementsHandler.runReadyTrigger` on the Elementor Elements to re-init them after the search has happened.

So far so good. 

However the issue is with the loop Carousel.  As soon as we replace the HTML in the carousel, we get a JS error:

![image](https://github.com/elementor/elementor/assets/853547/b4f622e8-b474-4585-b557-33478cbcfc44)

What I think we need is a way to stop/unload the JS events of the loop carousel.

### Describe the solution you'd like

That got me thinking, it would be really handy if an Element that has JS, always had an "unload" function (I don't see this in the docs) so it would be consistently implemented across all elements that need it.

**Comment by VladN06** [comment]
It would  be great to have this issue fixed  in Elementor. 
I've beed trying to use Search & Filter with the loop carousel and it would be a great feature.
